pixaris/generation/gemini.py

```diff
@@ -0,0 +1,257 @@
+import time
+import traceback
+from typing import List, Optional
+from pixaris.generation.base import ImageGenerator
+from PIL import Image
+from io import BytesIO
+import vertexai
+from google import genai
+from google.genai import types
+from vertexai.generative_models import Image as VertexImage
+
+
+class GeminiGenerator(ImageGenerator):
+    """
+    GeminiGenerator is a class that generates images using the Google Gemini API.
+
+    :param gcp_project_id: The Google Cloud Platform project ID.
+    :type gcp_project_id: str
+    :param gcp_location: The Google Cloud Platform location.
+    :type gcp_location: str
+    """
+
+    def __init__(self, gcp_project_id: str, gcp_location: str, verbose: bool = False):
+        self.gcp_project_id = gcp_project_id
+        self.gcp_location = "us-central1"  # For now since model only works there
+        self.verbose = verbose
+
+    def validate_inputs_and_parameters(
+        self,
+        dataset: List[dict[str, List[dict[str, Image.Image]]]] = [],
+        parameters: list[dict[str, str, any]] = [],
+    ):
+        """
+        Validates the provided dataset and parameters for image generation.
+
+        :param dataset: A list of datasets containing image and mask information.
```

Comment:
```suggestion
        :param dataset: A list of dicts containing image and mask information.
```
sonst verwirrend mit unserer Definition von dataset